common/taskana-common/src/main/java/pro/taskana/common/api/WorkingTimeCalculator.java

```diff
@@ -6,47 +6,161 @@
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
-import java.time.ZoneId;
+import java.time.ZoneOffset;
 import java.time.temporal.ChronoUnit;
 import java.util.EnumMap;
 import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
 
 import pro.taskana.common.api.exceptions.InvalidArgumentException;
 
 public class WorkingTimeCalculator {
 
-  public static final Map<DayOfWeek, LocalTimeInterval> WORKING_TIME;
+  static final ZoneOffset UTC = ZoneOffset.UTC;
 
-  static {
-    WORKING_TIME = new EnumMap<>(DayOfWeek.class);
-    WORKING_TIME.put(
-        DayOfWeek.MONDAY, new LocalTimeInterval(LocalTime.of(6, 0), LocalTime.of(18, 0)));
-    WORKING_TIME.put(
-        DayOfWeek.TUESDAY, new LocalTimeInterval(LocalTime.of(6, 0), LocalTime.of(18, 0)));
-    WORKING_TIME.put(
-        DayOfWeek.WEDNESDAY, new LocalTimeInterval(LocalTime.of(6, 0), LocalTime.of(18, 0)));
-    WORKING_TIME.put(
-        DayOfWeek.THURSDAY, new LocalTimeInterval(LocalTime.of(6, 0), LocalTime.of(18, 0)));
-    WORKING_TIME.put(
-        DayOfWeek.FRIDAY, new LocalTimeInterval(LocalTime.of(6, 0), LocalTime.of(18, 0)));
-    WORKING_TIME.put(DayOfWeek.SATURDAY, null);
-    WORKING_TIME.put(DayOfWeek.SUNDAY, null);
-  }
-
-  private final ZoneId zone;
   private final WorkingDaysToDaysConverter converter;
+  private final WorkingTimeSchedule workingTimeSchedule;
+
+  private final Map<DayOfWeek, LocalTimeInterval> legacyWorkingTimeSchedule;
```

Comment:
please remove legacy method and update release notes, so that we don't forget to communicate this